test/opt/instruction_test.cpp

```diff
@@ -168,7 +175,8 @@ TEST(InstructionTest, OperandConstIterators) {
 }
 
 TEST(InstructionTest, OperandIterators) {
-  Instruction inst(kSampleParsedInstruction);
+  IRContext context;
```

Comment:
Instead of having a default constructor that nobody else in the world uses, I recommend making a helper function in this module to make a dummy, but with few words.

e.g. 
static IRContext DummyContext() { return IRContext(Module(), ...); }

Then in the functions:
IRContext& context = DummyContext();

And then let lifetime extension carry it through.